### PR TITLE
WIP auth token

### LINKED ISSUE #310: New login method for authentication
The new login methods differ quite a bit from the old methods, first of all, there is now a method to authenticate without the use of csrf. The new method of authentication is fairly straightforward:

Firstly, do a `POST` request on `/api/v1/token` with the body containing the `'username'` and `'password'`. By doing so, you get a response with the newly acquired token in the `object`. After this, every request can be authenticated with the header parameter `X-Auth-Token`, granting access to the user. A token is valid for 5 days.

A logout functionality will be implemented in the future, this can create the possibility the token is expired even faster. We should take this into consideration with the new request methods.

### PR DESCRIPTION
**Pull request by TimvdLippe**
Have to go, but this is what I got so far. Probably a lot is still not working.

Fixes #310

### CODE REVIEW

**Comment by TimvdLippe** [comment]
@martijnjanssen Can you take this one?

File: src/lancie-content.html [hunk 1]
```diff
@@ -17,13 +17,7 @@
...
     </style>
 
-    <iron-ajax
-      auto
-      url="/api/v1/users/current"
-      handle-as="json"
-      on-response="storeUser"></iron-ajax>
-
-    <lancie-ajax id="ajax"></lancie-ajax>
+    <lancie-auth-storage id="auth" user={{user}}></lancie-auth-storage>
```
Comment (TimvdLippe) [inline src/lancie-content.html]: Forgot quotes around {{user}}

File: src/lancie-storage/lancie-auth-storage.html [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        // Check if token has expired
+        if (!this.auth || this.auth.created + 5 * 24 * 60 * 60 * 1000 < Date.now()) {
+          return;
+        }
+        return this.auth.token;
+      },
+
+      storeToken: function(token) {
+        this.set('auth', {
+          created: Date.now(),
```
Comment (TimvdLippe) [inline src/lancie-storage/lancie-auth-storage.html]: Isnt this created on the server?

File: src/lancie-storage/lancie-auth-storage.html [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          token: token,
+        });
+      },
+
+      storeUser: function(e, request) {
+        if (request.succeeded) {
```
Comment (TimvdLippe) [inline src/lancie-storage/lancie-auth-storage.html]: Show an error if it failed

File: src/lancie-storage/lancie-auth-storage.html (hunk 2 above)
Comment (martijnjanssen) [inline src/lancie-storage/lancie-auth-storage.html]: Oh yeah, you are correct, I was planning on changing that later but I forgot about it.

**Comment by TimvdLippe** [comment]
Branch is now working :tada:

**Comment by martijnjanssen** [comment]
Looks good!

**Pull request merged**